File: app/src/main/java/com/yuta/__old/ui/dialog/photo/DeletePhotoDialog.java
package com.yuta.__old.ui.dialog.photo;

import android.content.Context;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import com.yuta.__old.R;
import com.yuta.app.domain.model.response.UpdateResponse;
import com.yuta.common.ui.InteractiveDialog;
import com.yuta.__old.ui.fragment.ProfileFragment;
import com.yuta.app.network.RequestViewModel;
import com.yuta.common.util.UserUtils;

@SuppressWarnings("ConstantConditions")
public class DeletePhotoDialog extends InteractiveDialog {
    private RequestViewModel viewModel;

    public DeletePhotoDialog(Context context, Fragment fragment) {
        super(context, fragment);
        setDialogLayout(R.layout.dialog_delete);
    }

    @Override
    public void start() {
        super.start();
        viewModel = new ViewModelProvider(fragment).get(RequestViewModel.class);

        dialog.findViewById(R.id.close).setOnClickListener(v -> dismiss());
        dialog.findViewById(R.id.submit).setOnClickListener(v -> {
            deletePhoto(fragment);
            dismiss();
        });
    }

    protected void deletePhoto(Fragment fragment) {
        viewModel.getResultLiveData().removeObservers(fragment);
        viewModel.deleteUserPhoto(UserUtils.getCurrentUser().getId());
        viewModel.getResultLiveData().observe(fragment, result -> {
            if (!(result instanceof UpdateResponse)) return;
            ((ProfileFragment) fragment).updateProfile();
            dismiss();
        });
    }
}

File: app/src/main/java/com/yuta/profile/ui/dialog/DeletePhotoDialog.kt
package com.yuta.profile.ui.dialog

import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.viewModelScope
import com.yuta.app.R
import com.yuta.common.ui.CancelableDialog
import com.yuta.common.util.UserUtils
import com.yuta.profile.viewmodel.UserDetailsViewModel
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.launch

class DeletePhotoDialog(
    private val fragment: Fragment,
    private val onDeleteSuccessCallback: () -> Unit
) : CancelableDialog(R.layout.dialog_delete, fragment.requireActivity()) {

    private val closeButton: Button by lazy { dialog.findViewById(R.id.close) }
    private val submitButton: Button by lazy { dialog.findViewById(R.id.submit) }

    private val detailsViewModel: UserDetailsViewModel by fragment.viewModels()

    override fun start() {
        super.start()

        closeButton.setOnClickListener { dismiss() }
        submitButton.setOnClickListener { deletePhoto() }
    }

    private fun deletePhoto() {
        detailsViewModel.viewModelScope.launch {
            val isDeleted = detailsViewModel.deletePhoto(UserUtils.getUserId(fragment.requireContext())).first()
            handleDeleteResult(isDeleted)
        }
    }

    private fun handleDeleteResult(isDeleted: Boolean) {
        if (isDeleted) {
            onDeleteSuccessCallback()
            dismiss()
        }
    }
}
